### PR TITLE
RDX: Implement running commands in containers

### LINKED ISSUE #4022: RDX: Abstraction layer over moby/nerdctl for running a select set of commands
- [x] read file contents from a given image ID
- [x] copy files/directories to the host system, for a given image ID
- [x] run a container from a given image ID
- [x] stop the container
- [ ] execute a command inside a container
- [x] attach this abstraction layer to the backend

### LINKED ISSUE #4357: RDX: container API: execing into containers
We will need to implement APIs to allow extensions to spawn processes inside of their containers.

### PR DESCRIPTION
**Pull request by mook-as**
Since we unified to always using compose to run images, we can just look for a random service (i.e. container inside compose application) to run the commands on.

Fixes #4357 (implements the API).
Fixes #4022 (by finishing up the last item).

### CODE REVIEW

File: pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts [hunk 1]
```diff
@@ -280,28 +336,7 @@ export class NerdctlClient implements ContainerEngineClient {
...
-
-      return;
-    }
-
-    await this.nerdctl(...addNS('container', 'stop', container));
-    if (options?.delete) {
-      await this.nerdctl(...addNS('container', 'rm', container));
+      this.runCleanups(cleanups);
     }
   }
```
Comment (adamkpickering) [inline pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts:342]: We don't need to copy the compose file into the VM for doing `nerdctl compose down` do we? IIRC with `docker-compose` you need access to the compose file to do `docker-compose down`.

File: pkg/rancher-desktop/backend/containerClient/mobyClient.ts [hunk 2]
```diff
@@ -187,4 +189,22 @@ export class MobyClient implements ContainerEngineClient {
 
     console.debug('ran docker compose down', result);
   }
+
+  composeExec(composeDir: string, options: ContainerComposeExecOptions) {
```
[the remark this answers is not in the document]
Comment (adamkpickering) [reply pkg/rancher-desktop/backend/containerClient/mobyClient.ts:193]: Same comment for the `NerdctlClient.compose*` methods.

**Review by adamkpickering — CHANGES_REQUESTED** [review]

File: pkg/rancher-desktop/backend/containerClient/mobyClient.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/backend/containerClient/mobyClient.ts]: Explicit return types added in b4987fd8236cf99b83d8545c35aa135927103d3e

File: pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts (hunk 1 above)
Comment (mook-as) [reply pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts:343]: Good catch, the compose files are now copied over in b4987fd8236cf99b83d8545c35aa135927103d3e (it didn't make any sense to pass host-side `composeDir` in either way).

**Review by adamkpickering — APPROVED** [review]
I'm not sure that this is working properly with nerdctl, but approving to keep the ball rolling. I will make a comment below with a link to the relevant issue.

**Comment by adamkpickering** [comment]
The relevant issue is #4375.

**Pull request merged**